### DISCUSSION TITLE
[Feature] Support s3 compatible storage

### DISCUSSION THREAD

**Discussion by thino-dev** [immich-app/immich#11611] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Feature request:
Using localy storage have many limit like network bandwith, caching, word-wide user access. Make s3 compatible storage as option help image and video can load more fast. S3 compatible like CloudFlare R2 has CDN over world can reduce load of server and load time.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#1683
